### PR TITLE
Add UnsupportedOSPlatformAttribute to ProtectedBrowserStorage APIs

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs [hunk 1]
```diff
@@ -4,16 +4,19 @@
...
 using Microsoft.AspNetCore.DataProtection;
 using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.Web.Extensions
 {
+
     /// <summary>
     /// Provides mechanisms for storing and retrieving data in the browser storage.
     /// </summary>
+    [UnsupportedOSPlatform("browser")]
```
Comment (pranavkm) [inline src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs:19]: @jeffhandley am I holding this right? Initializing this type throws a PlatformNotSupportedException in Browser:

https://github.com/dotnet/aspnetcore/blob/master/src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs#L34-L37

Comment (jeffhandley) [reply src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs:19]: Yep; looks good!

**Review by jeffhandley — APPROVED** [review]

**Review by MackinnonBuck — APPROVED** [review]

File: src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedLocalStorage.cs [hunk 2]
```diff
@@ -15,6 +16,7 @@ namespace Microsoft.AspNetCore.Components.Web.Extensions
     ///
     /// See: https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage
     /// </summary>
+    [UnsupportedOSPlatform("browser")]
     public class ProtectedLocalStorage : ProtectedBrowserStorage
```
Comment (adamsitnik) [inline src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedLocalStorage.cs:20]: if `ProtectedBrowserStorage` has `[UnsupportedOSPlatform("browser")]` applied and `ProtectedLocalStorage` derives from it, then most probably there is no need to apply it to `ProtectedLocalStorage` and other deriving classes (`ProtectedSessionStorage` in this case)

@buyaa-n am I right?

Comment (buyaa-n) [reply src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedLocalStorage.cs:20]: The attributes are defined as [`Inherited = false`](https://github.com/dotnet/runtime/blob/06963eaedfe666c087a9fabb89b3f4bca57634c7/src/libraries/System.Private.CoreLib/src/System/Runtime/Versioning/PlatformAttributes.cs#L130). So i guess `ISymbol.GetAttrbiutes()` wouldn't return parent attributes and the analyzer not loading parent attributes manually. So guess we should attribute derived types

**Review by adamsitnik — APPROVED** [review]
LGTM!

File: src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedLocalStorage.cs (hunk 2 above)
Comment (adamsitnik) [reply src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedLocalStorage.cs:20]: @buyaa-n thanks for a great answer!

**Comment by mkArtakMSFT** [comment]
This has been reviewed and approved.

**Pull request merged**